javaee8/security-1.0/src/test/java/org/wildfly/swarm/ts/javaee8/security/CustomFormBasedAuthTest.java

```
@@ -0,0 +1,97 @@
+package org.wildfly.swarm.ts.javaee8.security;
+
+import java.io.File;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.drone.api.annotation.Drone;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.openqa.selenium.By;
+import org.openqa.selenium.NoSuchElementException;
+import org.openqa.selenium.WebDriver;
+import org.wildfly.swarm.ts.javaee8.security.customformauth.ApplicationConfig;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.jboss.shrinkwrap.api.ShrinkWrap.create;
+
+@RunWith(Arquillian.class)
+public class CustomFormBasedAuthTest {
+    @Deployment
+    public static WebArchive createDeployment() {
+        return create(WebArchive.class)
+                .addPackage(ApplicationConfig.class.getPackage())
+                .addAsWebResource(new File("src/main/webapp/login.xhtml"))
+                .addAsWebResource(new File("src/main/webapp/logout.xhtml"))
+                .addAsWebResource(new File("src/main/webapp/error.html"))
+                .addAsWebInfResource(new File("src/main/webapp/WEB-INF/jboss-web.xml"))
+                .addAsWebInfResource(new File("src/main/webapp/WEB-INF/faces-config.xhtml"));
+    }
+
+    @Drone
+    private WebDriver driver;
+
+    @After
+    public void close() {
+        try {
+            driver.get("http://localhost:8080/logout.xhtml");
+            driver.findElement(By.id("form:logout")).submit();
+            assertThat(driver.getPageSource()).contains("Login");
+        } catch (NoSuchElementException ex) {
```

Don't swallow exceptions.